packages/lit-dev-content/src/components/litdev-tutorial.ts

```diff
@@ -69,89 +93,217 @@ export class LitDevTutorial extends LitElement {
   @state()
   private _preloadedHtml?: {idx: number; promise: Promise<string>};
 
+  /**
+   * Tutorial manifest json file
+   */
+  @state()
+  private _manifest: TutorialManifest = {steps: [{title: ''}]};
+
   /**
    * Whether the tutorial is currently in its "solved" state.
    */
   private _solved = false;
 
+  private get _projectLocation() {
+    // TODO(e111077): delete this once we go public with tutorials catalog
+    if (window.location.pathname === '/tutorial/') {
```

Comment:
WDYT about using a similar URL-testing approach as this for the `use-old-url` and `show-catalog-button` options too? That would mean we don't need to add the new `view.html` file, which is basically identical to `tutorial/index.html` except that it changes those attributes. We could instead have the middleware serve `tutorial/index.html` for the new URL scheme. They wouldn't even need to be reactive properties, since they can't change.

Reply:
I really don't like url testing I personally would like to make every URL part user-inputted. I feel like it ties this component too close to this specific URL and section of this app and makes for potential refactors more painful as the URL dependencies are distributed across both frontend and backend. I'd rather just stick with this for now since we're going to remove it in a few weeks

Reply:
Sounds good.

> I personally would like to make every URL part user-inputted.

Can you clarify?

Reply:
basically make the element more generic and make every reference to every URL base an `@property`. like `/samples/js/tutorials/tutorial.json` could be `${this.jsSamplesPath}/${this.tutorialsPath}/tutorial.json` and same thing for lots of things in `_idxToInfo`